### PR TITLE
fix module not found in backend webpack tools

### PR DESCRIPTION
**Pull request by paulius-valiunas**
Webpacking backends results in this error:
`Module not found: Error: Can't resolve 'encoding' in 'D:\code\iModelJs\general-purpose-imodeljs-backend\node_modules\node-fetch\lib'`
`encoding` is an optional dependency so it should be ignored by webpack. 
`IgnoreOptionalDependenciesPlugin` didn't work so I went with a Stack Overflow suggestion to add it to "externals".

### CODE REVIEW

File: tools/backend-webpack/config/getWebpackConfig.js [hunk 1]
```diff
@@ -35,6 +35,9 @@ function getWebpackConfig(srcFile, outDir, profile) {
       // See https://github.com/facebook/create-react-app/issues/240
       new CaseSensitivePathsPlugin(),
     ],
+    externals: {
+      "encoding": "require('encoding')",
```
Comment (wgoehrig) [inline tools/backend-webpack/config/getWebpackConfig.js:39]: This should probably go here instead: https://github.com/iTwin/itwinjs-core/blob/release/2.19.x/tools/webpack-core/src/plugins/BackendDefaultsPlugin.ts#L68

**Review by wgoehrig — CHANGES_REQUESTED** [review]

File: tools/backend-webpack/config/getWebpackConfig.js (hunk 1 above)
Comment (wgoehrig) [reply tools/backend-webpack/config/getWebpackConfig.js:39]: Actually, try adding `"node-fetch"` to here instead: https://github.com/iTwin/itwinjs-core/blob/release/2.19.x/tools/webpack-core/src/plugins/BackendDefaultsPlugin.ts#L49

File: tools/webpack-core/src/plugins/BackendDefaultsPlugin.ts [hunk 2]
```diff
@@ -24,6 +24,11 @@ export class BackendDefaultsPlugin {
     // Don't bother minimizing backends...
     compiler.options.optimization!.minimize = false;
 
+    // optional dependency from node-fetch, doesn't work with IgnoreOptionalDependenciesPlugin
+    compiler.options.externals = {
```
Comment (calebmshafer) [inline tools/webpack-core/src/plugins/BackendDefaultsPlugin.ts:28]: Did neither of the changes t the webpack plugins work?

File: tools/backend-webpack/config/getWebpackConfig.js (hunk 1 above)
Comment (paulius-valiunas) [reply tools/backend-webpack/config/getWebpackConfig.js]: ooh, so I was adding the wrong package to the optional dependencies plugin... by the way, I find it weird that isn't the default for webpack. If someone wraps a require in a try/catch, it's obvious they _expect_ it to not always be available.

**Review by calebmshafer — APPROVED** [review]

**Review by wgoehrig — APPROVED** [review]

**Comment by calebmshafer** [comment]
@paulius-valiunas you'll need to run `rush change`

**Comment by paulius-valiunas** [comment]
> 
> 
> @paulius-valiunas you'll need to run `rush change`

I wish it worked properly on this branch :/

**Review by aruniverse — APPROVED** [review]

**Pull request merged**